.azure-pipelines/azure-pipelines-UpgrateVersion.yml

```diff
@@ -18,10 +18,25 @@ pool: sonicbld
 
 stages:
 - stage: Build
+  variables:
+    CACHE_MODE: none
+    VERSION_CONTROL_OPTIONS: 'SONIC_VERSION_CONTROL_COMPONENTS='
   jobs:
   - template: azure-pipelines-build.yml
     parameters:
-      buildSlave: y
+      buildOptions: '${{ variables.VERSION_CONTROL_OPTIONS }} SONIC_BUILD_JOBS=$(nproc) ENABLE_IMAGE_SIGNATURE=y'
+      preSteps:
+      - script: |
+          containers=$(docker container ls | grep "sonic-slave" | awk '{ print $1 }')
+          if [ ! -z "$containers" ]; then
+            docker container kill $containers || true
```

Comment:
>docker container kill $containers [](http://example.com/codeflow?start=12&length=33)

It's dangerous to kill containers if you are not sure about what they are doing. What happens if you don't kill? Could you be more conservative?

Reply:
azp only run one job on one agent. If there is a running container, it must be a  hunging container.
We need to clean all sonic-slave* images to cover the whole build process. It is by design.